client/src/utils/runInlineScripts.ts

```diff
@@ -1,20 +1,27 @@
 /**
  * Runs any inline scripts contained within the given DOM element or fragment.
  */
+const runScript = (script: HTMLScriptElement) => {
+  if (!script.type || script.type === 'application/javascript') {
+    const newScript = document.createElement('script');
+    Array.from(script.attributes).forEach((key) =>
+      newScript.setAttribute(key.nodeName, key.nodeValue || ''),
+    );
+    newScript.text = script.text;
+    script.replaceWith(newScript);
+  }
+};
+
 const runInlineScripts = (element: HTMLElement | DocumentFragment) => {
-  const scripts = element.querySelectorAll(
-    'script:not([src])',
-  ) as NodeListOf<HTMLScriptElement>;
-  scripts.forEach((script) => {
-    if (!script.type || script.type === 'application/javascript') {
-      const newScript = document.createElement('script');
-      Array.from(script.attributes).forEach((key) =>
-        newScript.setAttribute(key.nodeName, key.nodeValue || ''),
-      );
-      newScript.text = script.text;
-      script.replaceWith(newScript);
-    }
-  });
+  const selector = 'script:not([src])';
+  if (element instanceof HTMLElement && element.matches(selector)) {
+    runScript(element as HTMLScriptElement);
+  } else {
+    const scripts = element.querySelectorAll(
+      'script:not([src])',
+    ) as NodeListOf<HTMLScriptElement>;
```

Comment:
Since we've extracted the selector to a variable...

```suggestion
    const scripts = element.querySelectorAll<HTMLScriptElement>(selector);
```

Reply:
Oops, yes, that was indeed what I was aiming for :-) Thanks!